pkg/sql/information_schema.go

```diff
@@ -552,6 +588,25 @@ func forEachColumnInTable(
 	return nil
 }
 
+func forEachForeignKeyInTable(
+	table *sqlbase.TableDescriptor,
+	fn func(
+		*sqlbase.IndexDescriptor,
+		*sqlbase.ForeignKeyReference,
+	) error,
+) error {
+	for i := range table.Indexes {
```

Comment:
@dt I just added a new commit that uses `GetConstraintInfo`, which I'll squash if you like it. It was a little intrusive into `GetConstraintInfo` but it actually makes the `pg_constraint` stuff more readable, as it's now easier to see what differs between constraint kinds.